backend/app/anti_recommenders/open_ai/open_ai_normal_anti_recommender/open_ai_normal_anti_recommender.py

```diff
@@ -65,9 +65,9 @@ def generate_anti_recommendations(
     ) -> Generator[AntiRecommendation, None, None]:
         """Yield anti-recommendations of a given record key."""
 
-        self.build_chain()
-        query = self.create_query(record_key)
-        response = self.generate_response(query)
+        self._build_chain()
```

Comment:
We discussed the premature generalization here.

If you need to generalize the workflow, use first-class functions instead of downcalls.

That looks like:

Base class:
```
def _generate_anti_recommendations(self, record_key, callable_to_build_chain, callable_to_create_query, callable_to_generate_response):
   callable_to_build_chain()
   callable_to_create_query
   ...
```

Subclass:
```
def generate_anti_recommendations(self, record_key): # Override of public method
  self._generate_anti_recommendations(record_key, callable_to_build_query=...
```

The signature of callables is `typing.Callable` with generic parameters.